datasource/keyvaultsecret/data_acc_test.go

```diff
@@ -0,0 +1,124 @@
+package keyvaultsecret
+
+import (
+	"context"
+	_ "embed"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"regexp"
+	"testing"
+
+	"github.com/Azure/azure-sdk-for-go/services/keyvault/auth"
+	"github.com/Azure/azure-sdk-for-go/services/keyvault/v7.1/keyvault"
+	"github.com/Azure/go-autorest/autorest"
+	"github.com/hashicorp/packer-plugin-sdk/acctest"
+)
+
+//go:embed test-fixtures/template.pkr.hcl
+var testDatasourceBasic string
+
+func TestAccAzureKeyvaultSecret(t *testing.T) {
+	testEnv := "dev"
+	secret := &KeyvaultSecret{
+		Name:        "packer-datasource-keyvault-test-secret",
+		Value:       "this_is_the_packer_test_secret_value",
+		ContentType: "text/html",
+		Tags:        map[string]*string{"environment": &testEnv},
+	}
+
+	testCase := &acctest.PluginTestCase{
+		Name: "azure_keyvault_secret_datasource_basic_test",
+		Setup: func() error {
+			return secret.Create()
+		},
+		Teardown: func() error {
+			return secret.Delete()
+		},
+		Template: testDatasourceBasic,
+		Check: func(buildCommand *exec.Cmd, logfile string) error {
+			if buildCommand.ProcessState != nil {
+				if buildCommand.ProcessState.ExitCode() != 0 {
+					return fmt.Errorf("Bad exit code. Logfile: %s", logfile)
+				}
+			}
+
+			logs, err := os.Open(logfile)
+			if err != nil {
+				return fmt.Errorf("Unable find %s", logfile)
+			}
+			defer logs.Close()
+
+			logsBytes, err := ioutil.ReadAll(logs)
+			if err != nil {
+				return fmt.Errorf("Unable to read %s", logfile)
+			}
+			logsString := string(logsBytes)
+
+			valueLog := fmt.Sprintf("null.basic-example: secret value: %s", secret.Value)
+			contentTypeLog := fmt.Sprintf("null.basic-example: secret content_type: %s", secret.ContentType)
+			environmentLog := fmt.Sprintf("null.basic-example: secret environment: %s", *secret.Tags["environment"])
+
+			if matched, _ := regexp.MatchString(valueLog+".*", logsString); !matched {
+				t.Fatalf("logs doesn't contain expected secret value %q", logsString)
+			}
+			if matched, _ := regexp.MatchString(contentTypeLog+".*", logsString); !matched {
+				t.Fatalf("logs doesn't contain expected secret ContentType %q", logsString)
+			}
+			if matched, _ := regexp.MatchString(environmentLog+".*", logsString); !matched {
+				t.Fatalf("logs doesn't contain expected secret tag %q", logsString)
+			}
+
+			return nil
+		},
+	}
+	acctest.TestPlugin(t, testCase)
+}
+
+type KeyvaultSecret struct {
+	Name        string
+	Value       string
+	Id          string
+	ContentType string
+	Tags        map[string]*string
+
+	client keyvault.BaseClient
+}
+
+func getAuthorizer() (autorest.Authorizer, error) {
+	os.Setenv("AZURE_AD_RESOURCE", "https://vault.azure.net")
+	os.Setenv("AZURE_TENANT_ID", os.Getenv("PKR_VAR_tenant_id"))
```

Comment:
To simplify the acceptance testing setup the expected environment variables should be the same as expected by the current builder acceptance tests. This way you don't have to set anything, maybe other than the AZURE_AD_RESOURCE which looks like it can be a const for the test cases. 

https://github.com/hashicorp/packer-plugin-azure/blob/59806118c759250ad54715b6265e4b78b7dc5121/builder/azure/arm/builder_acc_test.go#L6

Reply:
After your last comment about the AZURE_* variables I understand why you have this setup. Instead of PKR_VAR_* I recommend using the environment variables already required for the the acceptance tests. For example

```
os.Setenv("AZURE_TENANT_ID", os.Getenv("ARM_TENANT_ID"))
os.Setenv("AZURE_CLIENT_ID", os.Getenv("ARM_CLIENT_ID"))
...
```